demo/dlb-init.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+ENABLED_DLB_PF_PCIIDS=${ENABLED_DLB_PF_PCIIDS:-270b 2710 2714}
+NUMVFS=${NUMVFS:-4}
+
+bind() {
+    BSF=$1
+    DRIVER=$2
+
+    LOOP=0
+    while [ "$LOOP" -lt  5 ]
+    do
+        if echo -n 2> /dev/null "$BSF" > /sys/bus/pci/drivers/"$DRIVER"/bind; then
+            break
+        else
+           LOOP=$(("$LOOP" + 1))
+           echo -n 2> /dev/null "$BSF" > /sys/bus/pci/devices/"$BSF"/driver/unbind
+            sleep 1
+        fi
+    done
+}
+
+unbind() {
+    BSF=$1
+    OLD_DRIVER=$2
+
+    LOOP=0
+    while [ "$LOOP" -lt  5 ]
+    do
+        if echo -n 2> /dev/null "$BSF" > /sys/bus/pci/drivers/"$OLD_DRIVER"/unbind;
+        then
+            break
+        else
+           LOOP=$(("$LOOP" + 1))
+            sleep 1
+        fi
+    done
+}
+
+#get dlb pf pci ids
+DEVS=$(for pf in $ENABLED_DLB_PF_PCIIDS; do lspci -n | grep -e "$pf" | grep -o -e "^\\S*"; done)
+for dev in $DEVS; do
+  DEVPATH="/sys/bus/pci/devices/0000:$dev"
+
+  #enable sriov_numvfs
+  unbind 0000:"$dev" dlb2
+  bind 0000:"$dev" dlb2
+  echo "$NUMVFS" | tee -a "$DEVPATH"/sriov_numvfs
+
+  for ((i=0;i<"$NUMVFS";i++));do
+    #unbind vf
+    vf=$(basename "$(realpath "$DEVPATH"/virtfn"$i")")
+    unbind "$vf" dlb2
+
+    #configure vf
+    echo $(($(cat "$DEVPATH"/total_resources/num_atomic_inflights) / "$NUMVFS")) | tee -a "$DEVPATH"/vf"${i}"_resources/num_atomic_inflights&&
+    echo $(($(cat "$DEVPATH"/total_resources/num_dir_credits) / "$NUMVFS")) | tee -a "$DEVPATH"/vf"${i}"_resources/num_dir_credits &&
+    echo $(($(cat "$DEVPATH"/total_resources/num_dir_ports) / "$NUMVFS")) | tee -a "$DEVPATH"/vf"${i}"_resources/num_dir_ports &&
+    echo $(($(cat "$DEVPATH"/total_resources/num_hist_list_entries) / "$NUMVFS")) | tee -a "$DEVPATH"/vf"${i}"_resources/num_hist_list_entries &&
+    echo $(($(cat "$DEVPATH"/total_resources/num_ldb_credits) / "$NUMVFS")) | tee -a "$DEVPATH"/vf"${i}"_resources/num_ldb_credits &&
+    echo $(($(cat "$DEVPATH"/total_resources/num_ldb_ports) / "$NUMVFS")) | tee -a "$DEVPATH"/vf"${i}"_resources/num_ldb_ports &&
+    echo $(($(cat "$DEVPATH"/total_resources/num_ldb_queues) / "$NUMVFS")) | tee -a "$DEVPATH"/vf"${i}"_resources/num_ldb_queues &&
+    echo $(($(cat "$DEVPATH"/total_resources/num_sched_domains) / "$NUMVFS")) | tee -a "$DEVPATH"/vf"${i}"_resources/num_sched_domains &&
+    echo $(($(cat "$DEVPATH"/total_resources/num_sn0_slots) / "$NUMVFS")) | tee -a "$DEVPATH"/vf"${i}"_resources/num_sn0_slots &&
+    echo $(($(cat "$DEVPATH"/total_resources/num_sn1_slots) / "$NUMVFS")) | tee -a "$DEVPATH"/vf"${i}"_resources/num_sn1_slots
```

Comment:
does this mean that we're assigning total amount of every resource to *every* VF ? I don't think VFs can share resources.

Reply:
It means total_resources of PF is divided into the number of vfs.

Reply:
ah, missed that you divide it to NUMVFS. Thanks for the explanations! Resolved.